app/src/app/settings/exportd-job-settings/exportd-job-settings-builder/exportd-job-sources-step/exportd-job-sources-step.component.ts

```diff
@@ -64,27 +67,30 @@ export class ExportdJobSourcesStepComponent implements OnInit {
   }
 
   @Input() public mode: CmdbMode;
-  public typeList: CmdbType[] = [];
+  public typeList: CmdbType[];
   public operators: any[] = ['!=', '=='];
   public sourcesForm: FormGroup;
   readonly SOURCES = 'sources';
 
-  constructor(private formBuilder: FormBuilder, private typeService: TypeService) {}
+  constructor(private formBuilder: FormBuilder, private typeService: TypeService) {
+    this.typeSubscription = new Subscription();
+    this.typeList = [];
+  }
 
-  ngOnInit() {
+  public ngOnInit(): void {
     this.sourcesForm = this.formBuilder.group({
       sources: this.formBuilder.array([this.createSource()])
     });
-    this.typeService.getTypeList().subscribe((value: CmdbType[]) => this.typeList = value);
+    this.typeSubscription = this.typeService.getTypeList().subscribe((value: CmdbType[]) => this.typeList = value);
 
     if (this.mode === CmdbMode.Create) {
       this.delCondition(0, 0);
     }
   }
 
-  public getFields(item: any) {
+  public getFields(publicID: any) {
     try {
-      return this.typeService.findType(item).fields;
+      return this.typeList.find(id => id.public_id === publicID).fields;
```

Comment:
@michael-batz In the end, this was the mistake. The list of the service was called, not the list of the component.